modules/json/src/smithy4s/json/internals/JsoniterCodecCompilerImpl.scala

```diff
@@ -19,27 +19,40 @@ package internals
 
 import smithy4s.HintMask
 import smithy4s.schema._
+import smithy4s.codecs.FieldSkipCompiler
 
 private[smithy4s] case class JsoniterCodecCompilerImpl(
     maxArity: Int,
-    explicitDefaultsEncoding: Boolean,
     flexibleCollectionsSupport: Boolean,
     infinitySupport: Boolean,
     preserveMapOrder: Boolean,
     hintMask: Option[HintMask],
     lenientTaggedUnionDecoding: Boolean,
-    lenientNumericDecoding: Boolean
+    lenientNumericDecoding: Boolean,
+    fieldSkipCompiler: FieldSkipCompiler
 ) extends CachedSchemaCompiler.Impl[JCodec]
     with JsoniterCodecCompiler {
 
   type Aux[A] = JCodec[A]
 
   def withMaxArity(max: Int): JsoniterCodecCompiler = copy(maxArity = max)
 
+  @deprecated(
+    message = "Use withFieldSkipCompiler instead",
+    since = "0.18.30"
+  )
   def withExplicitDefaultsEncoding(
-      explicitDefaultsEncoding: Boolean
+      explicitNulls: Boolean
```

Comment:
this should probably be renamed back, for source compatibility (and also is `explicitNulls` even a good name, if we're not changing the semantic?)

Reply:
not sure why it got renamed in the first place. Probably some metals quirk.